src/Console/Command/OutboxConsumeCommand.php

```diff
@@ -5,35 +5,122 @@
+        private readonly OutboxConsumer $consumer
+    ) {
         parent::__construct();
     }
 
     protected function configure(): void
     {
         $this
-            ->addOption('limit', null, InputOption::VALUE_REQUIRED, 'How many messages should be consumed in one run');
+            ->addOption(
+                'limit',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'How many messages should be consumed in one run (deprecated: use "message-limit" option)'
+            )
+            ->addOption(
+                'message-limit',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'How many messages should be consumed in one run',
+                100
+            )
+            ->addOption(
+                'run-limit',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'The maximum number of runs this command should execute',
+                1
+            )
+            ->addOption(
+                'memory-limit',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'How much memory consumption should the worker be terminated'
+            )
+            ->addOption(
+                'time-limit',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'What is the maximum time the worker can run in seconds'
+            )
+            ->addOption(
+                'sleep',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'How much time should elapse before the next job is executed in microseconds',
+                1000
+            );
     }
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $limit = InputHelper::nullableInt($input->getOption('limit'));
+        $legacyLimit = InputHelper::nullableInt($input->getOption('limit'));
+        $messageLimit = InputHelper::int($input->getOption('message-limit'));
+        $runLimit = InputHelper::nullableInt($input->getOption('run-limit'));
+        $memoryLimit = InputHelper::nullableString($input->getOption('memory-limit'));
+        $timeLimit = InputHelper::nullableInt($input->getOption('time-limit'));
+        $sleep = InputHelper::int($input->getOption('sleep'));
+
+        // legacy limit option to message-limit
+        if ($legacyLimit !== null) {
+            $messageLimit = $legacyLimit;
+        }
+
+        $logger = new ConsoleLogger($output);
+
+        $eventDispatcher = new EventDispatcher();
+        $eventDispatcher->addSubscriber(new StopWorkerOnSigtermSignalListener($logger));
+
+        if ($runLimit) {
+            if ($runLimit <= 0) {
+                throw new InvalidArgumentGiven($runLimit, 'null|positive-int');
+            }
+
+            $eventDispatcher->addSubscriber(new StopWorkerOnIterationLimitListener($runLimit, $logger));
+        }
+
+        if ($memoryLimit) {
```

Comment:
should we here also check for positive-int? And if yes, maybe it makes sense to move it to the InputHelper.

Reply:
This is a string and will be parsed to an integer in `Bytes::parseFromString`. And only positive numbers can be parsed.